lib/src/process_helper/apple.rs

```diff
@@ -1,166 +1,19 @@
+use log::warn;
 
-// We allow anyhow in here, as this is a module that'll be strictly used internally.
-// As soon as it's obvious that this is code is intended to be exposed to library users, we have to
-// go ahead and replace any `anyhow` usage by proper error handling via our own Error type.
-use anyhow::Result;
-use log::debug;
-use nix::{
-    sys::signal::{self, Signal},
-    unistd::Pid,
-};
+use darwin_libproc;
+use libc;
 
-use super::ProcessAction;
-use crate::network::message::Signal as InternalSignal;
-
-pub fn compile_shell_command(command_string: &str) -> Command {
-    let mut command = Command::new("sh");
-    command.arg("-c").arg(command_string);
-
-    command
-}
-
-fn map_action_to_signal(action: &ProcessAction) -> Signal {
-    match action {
-        ProcessAction::Pause => Signal::SIGSTOP,
-        ProcessAction::Resume => Signal::SIGCONT,
-    }
-}
-
-fn map_internal_signal_to_nix_signal(signal: InternalSignal) -> Signal {
-    match signal {
-        InternalSignal::SigKill => Signal::SIGKILL,
-        InternalSignal::SigInt => Signal::SIGINT,
-        InternalSignal::SigTerm => Signal::SIGTERM,
-        InternalSignal::SigCont => Signal::SIGCONT,
-        InternalSignal::SigStop => Signal::SIGSTOP,
-    }
-}
-
-/// Convenience wrapper around `send_signal_to_child` for raw unix signals.
-/// Its purpose is to hide platform specific logic.
-pub fn send_internal_signal_to_child(
-    child: &Child,
-    signal: InternalSignal,
-    send_to_children: bool,
-) -> Result<bool> {
-    let signal = map_internal_signal_to_nix_signal(signal);
-    send_signal_to_child(child, signal, send_to_children)
-}
-
-/// Convenience wrapper around `send_signal_to_child` for internal actions on processes.
-/// Its purpose is to hide platform specific logic.
-pub fn run_action_on_child(
-    child: &Child,
-    action: &ProcessAction,
-    send_to_children: bool,
-) -> Result<bool> {
-    let signal = map_action_to_signal(action);
-    send_signal_to_child(child, signal, send_to_children)
-}
-
-/// Send a signal to one of Pueue's child process handles.
-/// We need a special since we assume that there's also a `sh -c` around the actuall process.
-pub fn send_signal_to_child(
-    child: &Child,
-    signal: Signal,
-    _send_to_children: bool,
-) -> Result<bool> {
-    let pid = child.id();
-    // Send the signal to the shell, don't propagate to its children yet.
-    send_signal_to_process(pid, signal, false)?;
-
-    signal::kill(Pid::from_raw(pid.try_into().unwrap()), signal)?;
-    Ok(true)
-}
-
-/// This is a helper function to safely kill a child process.
-/// Its purpose is to properly kill all processes and prevent any dangling processes.
-pub fn kill_child(task_id: usize, child: &mut Child, _kill_children: bool) -> bool {
-    match child.kill() {
-        Err(_) => {
-            debug!("Task {task_id} has already finished by itself");
-            false
+pub fn get_process_group_pids(pgrp: libc::pid_t) -> Vec<libc::pid_t> {
```

Comment:
From what I can see, this is the only place `libc` is used for.
The `libc::pid_t` type a type alias to `i32` for both linux and darwin systems.
https://docs.rs/libc/latest/libc/type.pid_t.html

Couldn't we just go with `i32` and safe the explicit `libc` dependency?
If it cannot be avoided, this should also be exposed via one of our other dependencies.

I don't feel comfortable to directly work with `libc` when we're using abstraction libraries for the rest of our system interaction. This will only lead to further dependency bloat and incompatibilities in the long run.

Reply:
You are quite right, it's i32 on either Linux or Darwin. Only the `darwin_libproc` crate actually uses `libc::pid_t` here whereas the `procfs`  just uses `i32`, and so I should have looked at what `libc::pid_t` is defined as on darwin.